src/external/tsa.ts

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import fetch, { FetchInterface } from 'make-fetch-happen';
+import { ua } from '../util';
+import { checkStatus } from './error';
+
+export interface TimestampRequest {
+  artifact: string;
+  hashAlgorithm: string;
+  certificates?: boolean;
+  nonce?: number;
+  tsaPolicyOID?: string;
+}
```

Comment:
@malancas would love to get your eyes on this just to validate the "optionality" of the fields in the request.